aspnetcore/blazor/common/samples/3.x/BlazorServerEFCoreSample/BlazorServerDbContextExample/Pages/AddContact.razor

```diff
@@ -0,0 +1,113 @@
+﻿@page "/add"
+
+@inject IDbContextFactory<ContactContext> DbFactory
+@inject NavigationManager Nav
+@inject IPageHelper PageHelper
+
+@if (Contact != null)
+{
+    <ContactForm Busy="@Busy" 
+                   Contact="@Contact" 
+                   IsAdd="true"
+                   CancelRequest="Cancel"
+                   ValidationResult="@(async (success) => await ValidationResultAsync(success))" />
+}
+@if (Success)
+{
+    <br />
+    <div class="alert alert-success">The contact was successfully added.</div>
+}
+@if (Error)
+{
+    <br />
+    <div class="alert alert-danger">Failed to update the contact (@ErrorMessage).</div>
+}
+
+@code {
+    /// <summary>
+    /// New <see cref="Contact"/>.
+    /// </summary>
+    public Contact Contact { get; set; }
+
+    /// <summary>
+    /// <c>True</c> when an asynchronous operation is running.
+    /// </summary>
+    public bool Busy;
+
+    /// <summary>
+    /// <c>True</c> after successful add.
+    /// </summary>
+    public bool Success;
+
+    /// <summary>
+    /// <c>True</c> when an error occurred.
+    /// </summary>
+    public bool Error;
+
+    /// <summary>
+    /// The error message
+    /// </summary>
+    public string ErrorMessage = string.Empty;
+
+    /// <summary>
+    /// Start with fresh <see cref="Contact"/>.
+    /// </summary>
+    /// <returns>A <see cref="Task"/>.</returns>
+    protected override Task OnInitializedAsync()
+    {
+        Contact = new Contact();
+        return base.OnInitializedAsync();
+    }
+
+    /// <summary>
+    /// Respond to a forms submission.
+    /// </summary>
+    /// <param name="success"><c>True</c> when valid.</param>
+    /// <returns>A <see cref="Task"/>.</returns>
+    public async Task ValidationResultAsync(bool success)
+    {
+        if (Busy)
+        {
+            return;
+        }
+
+        if (!success)
+        {
+            Success = false;
+            Error = false;
+            return;
+        }
+
+        Busy = true;
+
+        using var context = DbFactory.CreateDbContext();
```

Comment:
This will be problematic because the SP will hold on to the context after disposal. The factory needs to create its own scope and have a way of getting rid of the context.

I think it might be best to change the factory to have a Create/Release method pair and handle the disposal inside Release (disposing of the internal scope and the DbContext) or create a Disposable result time inside the factory that you new up in order to return the context and the scope in a way that the scope and associated context get disposed when the result type gets disposed.

Reply:
See previous comments.